src/decorators/tigris-field.ts

```diff
@@ -0,0 +1,161 @@
+ * 									of schema field.
+ */
+export function Field(type: TigrisDataTypes, options?: EmbeddedFieldOptions): PropertyDecorator;
+
+/**
+ * Field decorator is used to mark a class property as Collection field. Only properties
+ * decorated with `@Field` will be used in Schema.
+ */
+export function Field(
+	typeOrOptions?: TigrisDataTypes | TigrisFieldOptions | EmbeddedFieldOptions,
+	options?: TigrisFieldOptions | EmbeddedFieldOptions
+): PropertyDecorator {
+	return function (target, propertyName) {
+		propertyName = propertyName.toString();
+		let propertyType: TigrisDataTypes | undefined;
+		let fieldOptions: TigrisFieldOptions;
+		let embedOptions: EmbeddedFieldOptions;
+
+		if (typeof typeOrOptions === "string") {
+			propertyType = <TigrisDataTypes>typeOrOptions;
+		} else if (typeof typeOrOptions === "object") {
+			if (isEmbeddedOption(typeOrOptions)) {
+				embedOptions = typeOrOptions as EmbeddedFieldOptions;
+			} else {
+				fieldOptions = typeOrOptions as TigrisFieldOptions;
+			}
+		}
+
+		if (typeof options === "object") {
+			if (isEmbeddedOption(options)) {
+				embedOptions = options as EmbeddedFieldOptions;
+			} else {
+				fieldOptions = options as TigrisFieldOptions;
+			}
+		}
+
+		// if type or options are not specified, infer using reflection
+		if (!propertyType) {
+			Log.info(`Using reflection to infer type of ${target.constructor.name}#${propertyName}`);
+			let reflectedType;
+			try {
+				reflectedType =
+					Reflect && Reflect.getMetadata
+						? Reflect.getMetadata("design:type", target, propertyName)
+						: undefined;
+				propertyType = ReflectedTypeToTigrisType.get(reflectedType.name);
+			} catch {
+				throw new ReflectionNotEnabled(target, propertyName);
+			}
+
+			// if propertyType is Array, subtype is required
+			if (propertyType === TigrisDataTypes.ARRAY && embedOptions?.elements === undefined) {
+				throw new IncompleteArrayTypeDefError(target, propertyName);
+			}
+
+			// if propertyType is still undefined, it probably is a typed object
+			if (propertyType === undefined) {
+				propertyType = TigrisDataTypes.OBJECT;
+				embedOptions = { elements: reflectedType };
+			}
+		}
+
+		if (!propertyType) {
+			throw new CannotInferFieldTypeError(target, propertyName);
+		}
+
+		// if propertyType is Array, subtype is required
+		if (propertyType === TigrisDataTypes.ARRAY && embedOptions?.elements === undefined) {
+			throw new IncompleteArrayTypeDefError(target, propertyName);
+		}
+
+		getDecoratorMetaStorage().fields.push({
+			name: propertyName,
+			type: propertyType,
+			isArray: propertyType === TigrisDataTypes.ARRAY,
+			target: target.constructor,
+			embedType: embedOptions?.elements,
+			arrayDepth: embedOptions?.depth,
+			schemaFieldOptions: fieldOptions,
+		} as FieldMetadata);
+	};
+}
+
+const ReflectedTypeToTigrisType: Map<string, TigrisDataTypes> = new Map([
+	["String", TigrisDataTypes.STRING],
```

Comment:
what about UUIDs, date-time? not sure if you are already handling it somewhere, or I  guess that must be explicitly decorated by the user?

Reply:
As of now, anything non-primitive needs to be explicitly specified. I will be looking into all the remaining types with respect to how deserialization would be handled case by case. That would be an easy add to this.

Reply:
sg